src/guardian/guardian.service.ts

```diff
@@ -96,10 +97,8 @@ export class GuardianService implements OnModuleInit {
     this.logger.log('New staking router state cycle start');
 
     try {
-      const {
-        elBlockSnapshot: { blockHash, blockNumber },
-        data: stakingModules,
-      } = await this.stakingRouterService.getStakingModules();
+      const { blockHash, blockNumber, stakingModulesData } =
```

Comment:
let's separate business logic from non-business.
Move all try content into separate function

Reply:
This function should only check if new block shuld be proceed or not

Reply:
I see it  like this
`handleNewBlock(){
    meta = this.keysApiService.getLatestMeta();
    if (this.blockGuardService.isNeedToProcessNewState(meta))
    { try {
         createAndSendDepositSignature()
    } except {}
}`